swan-lake/development-tutorials/test-document-the-code/ballerina-profiler.md

```diff
@@ -0,0 +1,365 @@
+---
+title: Ballerina Profiler
+description: Check out how you can run and do a CPU profile of a Ballerina program.
+keywords: ballerina runtime, troubleshoot, profiler, CPU profile, flame graph
+permalink: /learn/ballerina-profiler/
+active: ballerina-profiler
+---
+
+Improvements in your Ballerina code might be required to increase its runtime performance, minimize code execution times, increase throughput, and reduce latency. To do that, we need to identify performance bottlenecks, implications of concurrent executions, and find which code segments take longer to execute, and areas to improve performance. This can be done by using a profiler.
```

Comment:
```suggestion
Improvements in your Ballerina code might be required to increase its runtime performance, minimize code execution times, increase throughput, and reduce latency. To do that, you need to identify performance bottlenecks, implications of concurrent executions, which code segments take longer to execute, and areas to improve performance. This can be done by using a profiler.
```